TACC/Core-Portal
# task/FP-508: Data Files Filter

## Overview: ##

Adds a "File Filter" dropdown to the Data Files Searchbar. Also uses the `DataFilesSearchbar` component in Site Search.

Areas effective:

- All data files listings for all systems in the datafiles area, excluding the Shared Workspaces listing
- Public Data
- Site Search, excluding CMS search

## Related Jira tickets: ##

* [FP-508](https://jira.tacc.utexas.edu/browse/FP-508)

## Testing Steps: ##
**Testing areas**
- My Data, Community Data, Public Data, Shared Workspaces, Site Search, Public Data (https://cep.dev/public-data)

In each of the testing areas above, perform the following steps:
1. Switch through various filters and ensure the correct files are displayed, e.g. "Folders" should only display directories
2. Submit a search query and ensure the filter persists for the search results
3. Click to other Data Files areas and ensure the filter persists
4. Submit a Site Search and ensure the filter resets
5. Go back and forth from Public Data and ensure the filter resets
6. Ensure that the filter resets and shows all files if "Back to All Files" is clicked

## UI Photos:
- **Filter selected**
<img width="1358" alt="Screen Shot 2021-08-03 at 3 48 33 PM" src="https://user-images.githubusercontent.com/20326896/128084128-12ee9347-30a1-49a9-bae5-68e01f32a363.png">

- **Filter unselected**
<img width="1344" alt="Screen Shot 2021-08-03 at 3 48 21 PM" src="https://user-images.githubusercontent.com/20326896/128084129-bfd45e32-a6a8-4fc9-bde7-f40fa82af2f5.png">


## Notes: ##
- The [FileTypes.js](https://github.com/TACC/Core-Portal/blob/task/FP-508--data-files-filter/client/src/components/DataFiles/DataFilesSearchbar/FileTypes.js) file contains all of the associated extensions for each file type. If you have a suggestion for another extension to be added (or removed) from a file type category, please made it known!
- **Known Issue:** Switching in-between Site Search subsections with a filter will cause the app to crash. This should be fixed with https://github.com/TACC/Core-Portal/pull/453

To-Do:
- [ ] Apply active filter to backend search queries
